docs/proposals/node-attribute-labels.md

```diff
@@ -0,0 +1,119 @@
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/proposals/node-attribute-labels.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Export Node Attributes as Labels
+
+## Motivation
+
+Kubernetes assumes a homogeneous cluster of nodes and does not facilitate scheduling
+to a subset defined by one or more arbitrary attributes.  Labels may be applied
+manually to nodes through the api, but cannot be defined as part of the node configuration.
+One such use case for this would be hardening a set of nodes for security and restrict
+certain pods to these nodes for regulatory reasons.
+
+## Goals
+
+Allow Kubernetes administrators and cloud providers to easily configure arbitrary
+node attribute labels.  These labels would be available for specifying pod
+scheduling requirements.  Examples could include `docker.node.kubernetes.io/version=docker://1.5`,
+`aws.node.kubernetes.io/az=us-east-1`, or `k.node.kubernetes.io/version=1.1`
+
+## Non-Goals
+
+Supporting labels for consumable / finite attributes.  "Attributes"
+such as SSD and GPU are consumable and should be represented as generic resources.
+See [generic resources](../../docs/design/resources.md).
+
+## Proposal
+
+### Label Namespace
+
+Node attribute labels will live under the namespace `<some prefix>.node.kubernetes.io` and be required
+to match the glob `*.node.kubernetes.io/*`.  The name space `node.kubernetes.io` is reserved for use by
+kubernetes.
+
+### Plugins
+
+We will allow cloud providers and kubernetes administrators to specify custom node attribute
+labels through plugins.  Plugins will be checked at node start up time and published
+to the api server as part of the Node.ObjectMeta.Labels structure.  The plugin directory
+can be specified via the kubelet server command line argument `node-label-plugin-dir`.
+This flag specifies a directory containing one or more executables or json files.  This directory
+should NOT be mountable by containers via HostPath.  At kubelet initialization time these files
+will be used to apply additional labels to the node.  After start up, plugins will be polled periodically
+(every 10 minutes) in a background thread to update the labels sent as part of Node.ObjectMeta.Labels
+during health checks.  If there are errors while updating the labels, then no updates will be applied
+and the last good set will be used.  If there is an error during the first time plugins are checked,
+then no additional labels will be applied but the non-plugin specified node labels will still be used.
+
+## Plugin File Format
+
+### Json
+
+Json file plugins will be identified by having the extension '.json' and expected to be a map[string]string.
+e.g. a file named `label.json` with contents`{"a.node.kubernetes.io/l1":"v1", "a.node.kubernetes.io/l2":"v2"}`
+
+## Executable
```

Comment:
do we really need executables?  Can't we just tell sysadmin to make a cron job that spits output to a file in this dir?  Executables seems sort of heavy for this.  I could be swayed.